src/Psl/Filesystem/Internal/write_file.php

```diff
@@ -27,8 +27,8 @@
 function write_file(string $file, string $content, bool $append): void
 {
     if (Filesystem\exists($file)) {
-        Psl\invariant(Filesystem\is_file($file), '$file is not a file.');
-        Psl\invariant(Filesystem\is_writable($file), '$file is not writeable.');
+        Psl\invariant(Filesystem\is_file($file), Str\format('File "%s" is not a file.', $file));
+        Psl\invariant(Filesystem\is_writable($file), Str\format('File "%s" is not writeable.', $file));
```

Comment:
we shouldn't format these in place, instead pass it to invariant to do the formatting if the given fact is false.

```suggestion
        Psl\invariant(Filesystem\is_writable($file), 'File "%s" is not writeable.', $file);
```

Reply:
Sure, you are right. I missed `Psl\invariant` can format message itself.